### PR TITLE
chore: add os tag to agent config of pipelines

### PR DESCRIPTION
**Pull request by andreitraistaru**
## Reason

Clearly specify the OS of the agents on which PR pipeline should run. 

## License Acceptance

By submitting this pull request, I confirm that my contribution is made under
the terms of the Apache 2.0 license. For more information on following
Developer Certificate of Origin and signing off your commits, please check
[`CONTRIBUTING.md`][3].

## PR Checklist

- [ ] If a specific issue led to this PR, this PR closes the issue.
- [ ] The description of changes is clear and encompassing.
- [ ] Any required documentation changes (code and docs) are included in this PR.
- [ ] API changes follow the [Runbook for Firecracker API changes][2].
- [ ] User-facing changes are mentioned in `CHANGELOG.md`.
- [ ] All added/changed functionality is tested.
- [ ] New `TODO`s link to an issue.
- [ ] Commits meet [contribution quality standards](https://github.com/firecracker-microvm/firecracker/blob/main/CONTRIBUTING.md#contribution-quality-standards).

---

- [ ] This functionality can be added in [`rust-vmm`][1].

[1]: https://github.com/rust-vmm
[2]: https://github.com/firecracker-microvm/firecracker/blob/main/docs/api-change-runbook.md
[3]: https://github.com/firecracker-microvm/firecracker/blob/main/CONTRIBUTING.md

### CODE REVIEW

**Review by luminitavoicu — DISMISSED** [review]

File: .buildkite/common.py [hunk 1]
```diff
@@ -29,14 +31,15 @@ def group(label, command, instances, kernels, agent_tags=None, **kwargs):
...
-            agents = [f"instance={instance}", f"kv={kv}"] + agent_tags
-            step = {
-                "command": command,
-                "label": f"{label1} {instance} {kv}",
-                "agents": agents,
-                **kwargs,
-            }
-            steps.append(step)
+        for os in distros:
+            for kv in kernels:
```
Comment (pb8o) [inline .buildkite/common.py:35]: I think the os/kernel should go together; we will not run al2023 with 4.14. so we probably want sthg like a list of tuples:

```python
DEFAULT_PLATFORMS = [
  (al2, 4.14),
  (al2, 5.10),
  (al2023, 6.1),
]
```

But for now we can maybe just add `os=al2` in `pipeline_pr.py:94`, and that should apply to the whole pipeline?

File: .buildkite/pipeline_perf.py [hunk 2]
```diff
@@ -90,7 +90,7 @@ def build_group(test):
 if not args.instances:
     args.instances = DEFAULT_INSTANCES
 if not args.kernels:
-    args.kernels = DEFAULT_KERNELS
+    args.kernels = DEFAULT_PLATFORMS
```
Comment (pb8o) [inline .buildkite/pipeline_perf.py:93]: we may need to also change how we pass this parameter then... maybe we need to use `nargs=2` in the parameter definition?

File: .buildkite/common.py (hunk 1 above)
Comment (andreitraistaru) [reply .buildkite/common.py]: Should be fixed now.

File: .buildkite/pipeline_perf.py (hunk 2 above)
Comment (andreitraistaru) [reply .buildkite/pipeline_perf.py]: Should be fixed now.

**Review by luminitavoicu — APPROVED** [review]

**Review by pb8o — APPROVED** [review]

**Pull request merged**